Lib/test/test_unittest/testmock/testmock.py

```diff
@@ -1556,6 +1558,14 @@ def f(x=None): pass
             mock.assert_has_calls([call()])
         self.assertIsNone(cm.exception.__cause__)
 
+        with self.assertRaisesRegex(
+                AssertionError,
+                '^{}$'.format(
+                    re.escape('Calls not found.\n'
+                              'Expected: [call()]\n'
+                              '  Actual: []'))) as cm:
+            uncalled_mock.assert_has_calls([call()])
```

Comment:
why does this need to be a regex? Could this test be made simpler?

Reply:
It is a copy of a similar test few lines above, so it is OK.

Reply:
Cargo culting isn't okay from my perspective. I'd prefer tests to be as simple and clear as possible.
I come to this test and I'm left wondering "Why a regex? where's the pattern?".

Reply:
The other tests don't use the pattern matching abilities of regex either, it's just used a way of checking the contents of the message, so I don't think im doing anything different.

I could rewrite them to something like:
```python
with self.assertRaises(AssertionError) as cm:
    ...

self.assertEqual(str(cm.exception), "...")
```

but since the same pattern in other places in the file, if we're changing it in one place should we change it everywhere? That feels out of scope for this PR at least.

Reply:
I would write it like you just proposed (and rewrite other tests someday if there is an opportunity). This is not a big deal, but existing code is a bit ugly. There are only two occurrences of this pattern, both were introduced in 2180f6b058effbf49ec819f7cedbe76ddd4b700c, the code was less monstrous before this.

Reply:
Yeah, I forgot how limited the testing tools are in CPython itself :sigh:
Given the situation, I'm fine with how it is.

Reply:
I changed this test to remove the regex usage: https://github.com/python/cpython/pull/113573/commits/3d3bcf6b3d9bc740e50f553a550ca997c984551e